calico/felix/config.py

```diff
@@ -343,8 +351,18 @@ def _finish_update(self, final=False):
 
         self._validate_cfg(final=final)
 
-        # Now the config has been validated, generate the IPTables mark masks
-        # we'll actually use internally.
+        # Now calculate config options that rely on parameter validation.
+
+        # Determine the ETCD addresses to use.
+        endpoints = [x.strip() for x in self.ETCD_ENDPOINTS.split(",")]
+        if len(endpoints[0]) > 0:
+            self.ETCD_SCHEME = endpoints[0].split("://")[0]
+            self.ETCD_ADDRS = [e.split("://")[1] for e in endpoints]
```

Comment:
Just occurred to me, we should randomise the ordering.  When you're running at scale, it's important that not every felix fails over to the same etcd in order.